src/components/PresenceMap/component.js

```diff
@@ -0,0 +1,194 @@
+            const heading = marker.heading ? marker.heading : 0;
+
+            const markerOptions = {
+                position: marker.position,
+                map,
+                zIndex: Math.round(marker.position.lat * 100000),
+            };
+
+            let promImage = null;
+
+            if (marker.image) {
+                promImage = new Promise((resolve, reject) => {
+                    const result = getBase64Image(marker.image, heading);
+                    result
+                        .then(img => {
+                            markerOptions.icon = img;
+                            resolve(true);
+                        })
+                        .catch(error => {
+                            reject(`Error creating new image. Details: ${error}`);
+                        });
+                });
+            } else {
+                promImage = new Promise(resolve => resolve(true));
+            }
+
+            promImage
+                .then(() => {
+                    const markerMap = new window.google.maps.Marker(markerOptions);
+                    markersCopy.push(markerMap);
+
+                    if (center === 'auto') {
+                        const bounds = new window.google.maps.LatLngBounds();
+                        markersCopy.forEach(markerObj => bounds.extend(markerObj.getPosition()));
+                        map.setCenter(bounds.getCenter());
+                        map.fitBounds(bounds);
+                    }
+
+                    setMap(map);
+                })
+                .catch(error => {
+                    // eslint-disable-next-line no-console
+                    console.log(`Error. Details: ${error}`);
+                });
+        });
+    };
+
+    return (
+        <StyledContainer style={style} ref={container}>
+            <StyledMapContainer
+                ref={mapContainer}
+                style={getMapContainerStyles(container.current)}
+            />
+        </StyledContainer>
+    );
+}
+
+MapComponent.propTypes = {
+    isScriptLoaded: PropTypes.bool.isRequired,
+    isScriptLoadSucceed: PropTypes.bool.isRequired,
+    objects: PropTypes.arrayOf(
+        PropTypes.shape({
+            position: PropTypes.shape({
+                lat: PropTypes.number,
+                lng: PropTypes.number,
+            }),
+            heading: PropTypes.number,
+            image: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+            onClick: PropTypes.func,
+        }),
+    ),
+    style: PropTypes.object,
+    zoom: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    center: PropTypes.oneOfType([
+        PropTypes.shape({
+            lat: PropTypes.number,
+            lng: PropTypes.number,
+        }),
+        PropTypes.string,
+    ]),
+    showTraffic: PropTypes.bool,
+    showTransit: PropTypes.bool,
+    // eslint-disable-next-line
+    children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
```

Comment:
children are not being used in this component, we need to pass this prop in order to render things inside the map